### ISSUE TITLE
joining react-admins ArrayReference

### ISSUE THREAD

**Issue by nkappler** [mevdschee/ra-data-treeql#4]
I am a bit stuck at the moment:

React Admin supports putting some JSON into the database. I know this is not the preferred way to store relational data, but if you don't want to subscribe to the premium plan of RA (which is 135€ per **month**!) or implement relational UI controls yourself from scratch, this is pretty much the only way to go, at least as far as I can tell at the moment.

I don't see a possibility for joining those in one request, do you have any ideas? In my case, the joined data will also contain foreign keys, so it will be a triple join, hence I don't want to send a bunch of requests. I also don't see a way to create a VIEW for this, but I am pretty bad at SQL 🤣

tldr, this results in either an array of indices stored in a single column or it could be any arbitrary object, which can also contain foreign keys (the latter one can be seen in the example below).

Example request:

```domain.com/api/index.php/records/product/5```

will return something similar to: 

```JSON
{
  "id":5,
  "name":"Product Name",
  "desc":null,
  "price":"null",
  "classes" :[
    {"class_id": 16},
    {"class_id": 11},
    {"class_id": 8}
  ]
}
```

How can I join those IDs in one request?

*Also, another thing I noticed and don't know where to put it:
Look at the "price" column, it has also data type JSON as it will contain some formula in the future,
but for some reason it returns "null" as string value instead of as a primitive. Is this a bug with the php-crud-api?
Afaik JSON will be parsed usually, but in this case I run into an error further down the line in react admin...*

**Comment by mevdschee** [comment]
Hi, thank you for asking this question. I think it should be

    domain.com/api/index.php/records/classes/16,11,8&join=products

I hope this helps!

**Comment by mevdschee** [comment]
>but for some reason it returns "null" as string value instead of as a primitive. Is this a bug with the php-crud-api?

This may indeed be a bug and should have been resolved with the JSON middleware. Handling things other than objects and arrays in JSON fields is not supported currently.

**Comment by nkappler** [comment]
> Hi, thank you for asking this question. I think it should be
> 
> ```
> domain.com/api/index.php/records/classes/16,11,8&join=products
> ```
> 
> I hope this helps!

no, unfortunately not. 
The `classes`' IDs are stored in the `product` table and there is no classic cross-table for defining an n-to-n relationship, so I don't know the IDs I need to fetch for a certain `product` beforehand, I first need to fetch the product and would like to join in the required `classes`, too.

The simplest possible `product` could also look like this, maybe it's possible to create a join for this?
```JavaScript
{
  id: 5,
  name: "product",
  classes: [0, 8, 15, 42],
  ...
}
```

**Comment by nkappler** [comment]
> ```js
> {
>   id: 5,
>   name: "product",
>   classes: [0, 8, 15, 42],
>   ...
> }
> ```

Would it help to rename `classes` column to `classes_id`? Are arrays supported with php-crud-api?

**Comment by mevdschee** [comment]
>Are arrays supported with php-crud-api?

No, by default they are not. They are if  you use json strings and the json middleware.

**Comment by nkappler** [comment]
Closing this, since it is not really an issue with the data provider...
